packages/vite/src/node/ssr/ssrExternal.ts

```diff
@@ -53,15 +62,7 @@ export function resolveSSRExternal(
     if (!entry.includes('node_modules')) {
       // entry is not a node dep, possibly linked - don't externalize
       // instead, trace its dependencies.
-      const depRoot = path.dirname(resolveFrom(`${id}/package.json`, root))
-      resolveSSRExternal(
-        {
-          ...config,
-          root: depRoot
-        },
-        knownImports,
-        ssrExternals
-      )
+      depsToTrace.add(id)
```

Comment:
Do we still need this change then?

Reply:
Yes, we need to ensure all the deps are visited and added to `seen` set before the recursive calls.